data/gen-tifu.py

```diff
@@ -0,0 +1,42 @@
+import os
+import io
+import json
+import sys
+import tqdm
+import gdown
+import random
+import subprocess
+from src.helpers import clean
+
+DATA_URL = "https://drive.google.com/uc?export=download&id=1swS7fuzE_UMhKtcWIJdOvbgBWl8cyGjd"
+IN = "tifu"
+OUT = "../datasets/tifu"
+PREFIX = "tifu"
+if len(sys.argv) == 1: random.seed(2022)
+else: random.seed(int(sys.argv[1]))
+
+if not os.path.exists("tifu_datasets.zip"):
+    gdown.download(DATA_URL, "tifu_datasets.zip", quiet=False)
+if not os.path.exists(IN):
+    os.mkdir(IN)
+    subprocess.call(['unzip', '-o', "tifu_datasets.zip", '-d', "../data/tifu"])
+
+files = os.listdir(IN)
+splits = ["train", "validation"]
+try: os.makedirs(OUT)
+except FileExistsError: pass
+outputs={split:io.open(os.path.join(OUT,f"{PREFIX}.{split}"), mode="w", encoding="utf-8") for split in splits}
+
+count = 0
+for filename in tqdm.tqdm(files):
+    with io.open(os.path.join(IN,filename), mode="r", encoding="utf8") as f:
+        for line in f:
+            data = json.loads(line)

```

Comment:
Instead of loading every line you can do` json.load(open("path-to-file", "r"))` to get the entire document (it's small enough to be 100% loaded into memory) Following that, you can iterate though the loaded object instead of blindly reading lines

Reply:
There seem to be some garbage between the lines, which doesn't allow me to load the entire thing, so I decided to load it line by line to work around it. I tried to dig around for a bit but can't find the exact problem or the possible solution, any idea what I could try? Or if we can leave it as it is.

Reply:
Hm
Yeah not a optimal solution; what's the error when loaded in its entirety?

Reply:
```
raise JSONDecodeError("Extra data", s, end)
json.decoder.JSONDecodeError: Extra data: line 2 column 1 (char 2176)
```
This ^^

Reply:
Looks to be https://stackoverflow.com/questions/11639886/how-to-read-a-json-file-containing-multiple-root-elements
No idea why people would do this when they could just make multiple json files
Should be solvable with some regex